server/datasources/requestTypeDataSource.ts

```diff
@@ -1,9 +1,80 @@
 import CachedMongooseDataSource from './cachedMongooseDataSource'
```

Comment:
Moved the softDelete function.

Reply:
I personally think it's not a good design choice to move softDeleteRequestType to cachedMongooseDataSource because it cannot be generically used by all data sources.

Reply:
Every cachedMongooseDataSource uses soft deletion except for clients (and they probably should). @meganniu, should we add a deleted field to client objects? 

If so, then the behaviour is the same for all data sources. If not, then we should at least write a typescript generic function for softDeletion since behaviour is the same for three data sources (requests, requestTypes, requestGroups).

Reply:
Yes let's add delete to client. we should be able to perform CRUD on all objects

Reply:
Oh also, i don't think we should make `create` and `update` functions generic
We need to perform type-specific validation and manipulation before creating/updating (e.g. check RequestGroup with id = `requestType.requestGroup` exists, remove RequestType id from old ReqestGroup when updating)

Reply:
Actually maybe we could make `create` and `update` function generic.
We pass a function that performs type-specific validation and manipulation into `create` and `update`
```
create(object, validate: (object) => boolean) {
  validate(object)
  create object
}
```

@guy-ro @kevinkwchan @Anand-Issac thoughts?

Reply:
We can also put the logic in the resolver instead of in datasources. That way datasources are only involved in making queries to DB, not making multiple queries to multiple collections.

Reply:
^ i like the idea of putting the logic in resolvers better. 
@kevinkwchan @Anand-Issac how does that sound for you guys?

Reply:
> I personally think it's not a good design choice to move softDeleteRequestType to cachedMongooseDataSource because it cannot be generically used by all data sources.

jk